bin/seedUserStreaks.ts

```diff
@@ -0,0 +1,168 @@
+import fastq from 'fastq';
+import { differenceInDays } from 'date-fns';
+
+import '../src/config';
+import createOrGetConnection from '../src/db';
+import { User, UserStreak, View } from '../src/entity';
+import { DataSource, DeepPartial, Equal, IsNull, MoreThan, Not } from 'typeorm';
+import { shouldResetStreak } from '../src/common';
+
+const QUEUE_CONCURRENCY = 1;
+const BATCH_SIZE = 1000;
+
+interface UserPointer {
+  id?: string;
+  createdAt?: string;
+}
+
+const addNextBatch = async (
+  con: DataSource,
+  insertQueue: fastq.queueAsPromised<Partial<User>[], void>,
+  userPointer?: UserPointer,
+  limit: number = BATCH_SIZE,
+): Promise<UserPointer> => {
+  const qb = con
+    .getRepository(User)
+    .createQueryBuilder('u')
+    // fetching createdAt as a string, because Date object in node is milisecond precision, but postgres is microsecond
+    .select(['id', 'timezone', '"createdAt"::text'])
```

Comment:
The devil is in the microsecond precision part. 

If we run the query without the text conversion, [typeorm messes up the microsecond precision and rounds up to the nearest milisecond instead](https://github.com/typeorm/typeorm/issues/4964). This is caused by the pg driver that typeorm uses. 

Which on my local env means, that the script keeps running forever, because:
1. user has `createdAt` set to `2024-02-15 19:24:19.474025`
2. returned value for `createdAt` is `2024-02-15 19:24:19.474Z` -> the `025` portion is rounded down
3. When running the where condition, `createdAt` > `2024-02-15 19:24:19.474Z` it succeeds, since `2024-02-15 19:24:19.474025 > 2024-02-15 19:24:19.474`

Reply:
Actually good that you caught it even. 👍